### PR TITLE
Test that HTMLInputElement.labels is null when type=hidden, but [SameObject] otherwise

### PR DESCRIPTION
**Pull request by cvrebert**
Refs https://github.com/whatwg/html/issues/2280

### CODE REVIEW

**Comment by w3c-bots** [comment]
# Chrome (unstable channel) #
Testing web-platform-tests at revision e777058b30f21fe28a1550458ef0779be52c4635
Using browser at version 58.0.3004.3 dev
Starting 10 test iterations
All results were stable
## All results ##
<details>
<summary><a href="http://w3c-test.org/submissions/4804/html/semantics/forms/the-label-element/labelable-elements.html">/html/semantics/forms/the-label-element/labelable-elements.html</a></summary>


|                              Subtest                               | Results |
|--------------------------------------------------------------------|---------|
|                                                                    | OK      |
| `Check if the img element is not a labelable element`              | PASS    |
| `Check if the select element is a labelable element`               | PASS    |
| `Check if the hidden input element has null 'labels'`              | PASS    |
| `Check if the object element can access 'labels'`                  | PASS    |
| `Check if the meter element can access 'labels'`                   | PASS    |
| `Check if the textarea element can access 'labels'`                | PASS    |
| `Check if the progress element is a labelable element`             | PASS    |
| `Check if the button element can access 'labels'`                  | PASS    |
| `Check if the input element in radio state can access 'labels'`    | PASS    |
| `Check if the input element in radio state is a labelable element` | PASS    |
| `Check if the label element can access 'labels'`                   | PASS    |
| `Check if the textarea element is a labelable form-element`        | PASS    |
| `Check if the meter element is a labelable element`                | PASS    |
| `Check if the hidden input element is not a labelable element.`    | PASS    |
| `Check if the button element is a labelable element`               | PASS    |
| `Check if the fieldset element is not a labelable element`         | PASS    |
| `Check if the keygen element is not a labelable element`           | PASS    |
| `Check if the fieldset element can access 'labels'`                | PASS    |
| `Check if the keygen element can access 'labels'`                  | PASS    |
| `Check if the output element is a labelable element`               | PASS    |
| `Check if the progress element can access 'labels'`                | PASS    |
| `Check if the img element can access 'labels'`                     | PASS    |
| `Check if the object element is not a labelable element`           | PASS    |
| `Check if the select element can access 'labels'`                  | PASS    |
| `Check if the label element is not a labelable element`            | PASS    |
| `Check if the output element can access 'labels'`                  | PASS    |

</details>

**Comment by w3c-bots** [comment]
# Firefox (nightly channel) #
Testing web-platform-tests at revision e777058b30f21fe28a1550458ef0779be52c4635
Using browser at version BuildID 20170123125947; SourceStamp 36486fdc3813ef7943ae5b07b4128866d1938a6c
Starting 10 test iterations
All results were stable
## All results ##
<details>
<summary><a href="http://w3c-test.org/submissions/4804/html/semantics/forms/the-label-element/labelable-elements.html">/html/semantics/forms/the-label-element/labelable-elements.html</a></summary>


|                              Subtest                               | Results |
|--------------------------------------------------------------------|---------|
|                                                                    | OK      |
| `Check if the img element is not a labelable element`              | PASS    |
| `Check if the select element is a labelable element`               | PASS    |
| `Check if the hidden input element has null 'labels'`              | FAIL    |
| `Check if the object element can access 'labels'`                  | PASS    |
| `Check if the meter element can access 'labels'`                   | FAIL    |
| `Check if the textarea element can access 'labels'`                | FAIL    |
| `Check if the progress element is a labelable element`             | PASS    |
| `Check if the button element can access 'labels'`                  | FAIL    |
| `Check if the input element in radio state can access 'labels'`    | FAIL    |
| `Check if the input element in radio state is a labelable element` | PASS    |
| `Check if the label element can access 'labels'`                   | PASS    |
| `Check if the textarea element is a labelable form-element`        | PASS    |
| `Check if the meter element is a labelable element`                | PASS    |
| `Check if the hidden input element is not a labelable element.`    | PASS    |
| `Check if the button element is a labelable element`               | PASS    |
| `Check if the fieldset element is not a labelable element`         | PASS    |
| `Check if the keygen element is not a labelable element`           | FAIL    |
| `Check if the fieldset element can access 'labels'`                | PASS    |
| `Check if the keygen element can access 'labels'`                  | PASS    |
| `Check if the output element is a labelable element`               | PASS    |
| `Check if the progress element can access 'labels'`                | FAIL    |
| `Check if the img element can access 'labels'`                     | PASS    |
| `Check if the object element is not a labelable element`           | PASS    |
| `Check if the select element can access 'labels'`                  | FAIL    |
| `Check if the label element is not a labelable element`            | PASS    |
| `Check if the output element can access 'labels'`                  | FAIL    |

</details>

File: html/semantics/forms/the-label-element/labelable-elements.html [hunk 1]
```diff
@@ -91,8 +91,23 @@
...
+  });
+
+  hiddenInput.type = "text";
+  testLabelsAttr("testHidden", "lbl5");
+  var labels = hiddenInput.labels;
+
+  hiddenInput.type = "hidden";
+  assert_equals(labels.length, 0, "Retained .labels NodeList should be empty after input type changed to hidden");
+
+  hiddenInput.type = "text";
```
Comment (zcorpan) [inline html/semantics/forms/the-label-element/labelable-elements.html]: Can you change this to `checkbox` and add an assertion for `labels.length` after (should be 1 but I seem to get 0 in chromium and webkit) ?

Comment (cvrebert) [inline html/semantics/forms/the-label-element/labelable-elements.html]: Done.

**Review by zcorpan — APPROVED** [review]

**Pull request merged**